ghost dependency domhandler

https://github.com/hexojs/hexo-util/blob/1cf5b21ceeb0d2f9456f9003b9f956c43903d4ff/lib/toc_obj.ts#L3

`domhandler` is not defined in `package.json` but used in code.


### build failed

```sh
 % pnpm tsc -b -f -v
[下午10:30:39] Projects in this build: 
    * tsconfig.json

[下午10:30:39] Project 'tsconfig.json' is being forcibly rebuilt

[下午10:30:39] Building project 'D:/hexo-util/tsconfig.json'...

lib/toc_obj.ts:3:30 - error TS2307: Cannot find module 'domhandler' or its corresponding type declarations.

3 import type { Element } from 'domhandler';
                               ~~~~~~~~~~~~


Found 1 error.
```

### eslint & ts-check

![vscode](https://github.com/hexojs/hexo-util/assets/22849383/dc375349-04ee-42f1-b94c-7049772227aa)

--- a/package.json
+++ b/package.json
@@ -40,6 +40,7 @@
     "c8": "^9.1.0",
     "chai": "^4.3.6",
     "chai-as-promised": "^7.1.1",
+    "domhandler": "^5.0.3",
     "eslint": "^8.23.0",
     "eslint-config-hexo": "^5.0.0",
     "html-entities": "^2.3.3",

